src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,120 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents the storage functionalities used by
+ * the Duke chat bot. It manages the loading and saving
+ * of tasks in and out of the disk.
+ */
+public class Storage {
+
+    protected Path filePath;
+
+    /**
+     * Initialises the class with a file path
+     * where the file will be used for data storage
+     * operations.
+     */
+    public Storage(String filePath) {
+        this.filePath = Paths.get(System.getProperty("user.dir"), filePath);
+    }
+
+    /**
+     * Loads task list data found in the disk
+     * into the Duke chat bot.
+     *
+     * @return Task list.
+     * @throws DukeException If the file cannot be found.
+     */
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> collection = new ArrayList<>();
+
+        List<String> lines;
+        try {
+            lines = Files.readAllLines(this.filePath);
+            for (String line : lines) {
+                String[] parts = line.replace('|', ' ').split("[ ]+");
+
+                Task task;
+                if (parts[0].equals("T"))
```

Comment:
What is parts[0]? Abstract more
e.g. String command = parts[0];
if (command.equals("T")) {
    ...
}

Reply:
Also missing curly bracket